src/components/capture/create/DetailsForm.tsx

```diff
@@ -1,70 +1,69 @@
-import useConnectors from 'hooks/useConnectors';
-import { useEffect, useState } from 'react';
+import { useMemo } from 'react';
 import { FormattedMessage, useIntl } from 'react-intl';
 import {
     defaultOptions,
     defaultRenderers,
     showValidation,
 } from 'services/jsonforms';
 
-type NewCaptureDetailsProps = {
+interface NewCaptureDetailsProps {
     displayValidation: boolean;
     readonly: boolean;
-};
+    connectorTags: any[];
+}
 
 const stateSelectors = {
     formData: (state: CaptureCreationState) => state.details.data,
     setDetails: (state: CaptureCreationState) => state.setDetails,
-    setHasConnectors: (state: CaptureCreationState) => state.setHasConnectors,
+    setConnectors: (state: CaptureCreationState) => state.setConnectors,
 };
 
 function NewCaptureDetails(props: NewCaptureDetailsProps) {
-    const intl = useIntl();
-    const { readonly, displayValidation } = props;
+    const { readonly, displayValidation, connectorTags } = props;
 
+    const intl = useIntl();
     const formData = useCaptureCreationStore(stateSelectors.formData);
     const setDetails = useCaptureCreationStore(stateSelectors.setDetails);
-    const setHasConnectors = useCaptureCreationStore(
-        stateSelectors.setHasConnectors
-    );
-
-    const {
-        data: connectorsData,
-        isError,
-        isSuccess,
-        error,
-    } = useConnectors(ConnectorTypes.SOURCE);
 
-    const [isPostProcessingDone, setPostProcessingDone] = useState(false);
-    const [schema, setSchema] = useState({
-        properties: {
-            image: {
-                description: intl.formatMessage({
-                    id: 'captureCreation.image.description',
-                }),
-                oneOf: [] as { title: string; const: string }[],
-                type: 'string',
+    const schema = useMemo(() => {
+        return {
+            properties: {
+                image: {
+                    description: intl.formatMessage({
+                        id: 'captureCreation.image.description',
+                    }),
+                    oneOf:
+                        connectorTags.length > 0
+                            ? connectorTags.map((connector: any) => {
+                                  return {
+                                      const: connector.id,
+                                      title: connector.connectors.image_name,
+                                  };
+                              })
+                            : ([] as { title: string; const: string }[]),
+                    type: 'string',
+                },
+                name: {
+                    description: intl.formatMessage({
+                        id: 'captureCreation.name.description',
+                    }),
+                    maxLength: 1000,
+                    minLength: 3,
+                    pattern: '^[a-zA-Z0-9_.-]*/[a-zA-Z0-9_.-]+$',
+                    type: 'string',
+                },
```

Comment:
Instead of handling stuff "after the server call" we are just memoizing the schema and passing the connectors from the parent.